### PR TITLE
cpu/fe310: slightly rework the uart driver

### PR DESCRIPTION
**Pull request by aabadie**
### Contribution description

While working on #12891, I noticed that the UART driver of the fe310 CPU would need some improvement:
- not inline with the usual way of configuring peripheral at board level
- code duplication
- gpio used by UART initialized in board_init

~This PR fixes all these problems while keeping the UART working. This PR is also saving some ROM memory (40B):~
Strange, the ROM size is increased now. Need to investigate...

- This PR:

```
   text	   data	    bss	    dec	    hex	filename
  10628	    112	   2408	  13148	   335c	/work/riot/RIOT/examples/hello-world/bin/hifive1b/hello-world.elf
```

- master:

```
   text	   data	    bss	    dec	    hex	filename
  10592	    112	   2408	  13112	   3338	/work/riot/RIOT/examples/hello-world/bin/hifive1b/hello-world.elf
```

### Testing procedure

- A green CI
- Run an application with the shell, it should still work

### Issues/PRs references

None

### CODE REVIEW

**Comment by kaspar030** [comment]
Force pushing causes github notification links to go stale. :(
Do you mind going with regular fixup commits?

File: boards/hifive1b/include/periph_conf.h [hunk 1]
```diff
@@ -28,7 +30,7 @@ extern "C" {
  * @{
  */
 /* As defined in boards/hifive1/board.c CPU_DESIRED_FREQ **/
```
Comment (kaspar030) [inline boards/hifive1b/include/periph_conf.h]: is the comment correct?

Comment (aabadie) [reply boards/hifive1b/include/periph_conf.h]: Indeed, haven't checked the comment. Anyway, I have a branch where I rework the file organization more in depth, and this comment is gone there. It still needs a bit of work, but I'll PR it soon.

**Comment by aabadie** [comment]
> Do you mind going with regular fixup commits?

Sorry, some racy workflow: since I got no review yet and noticed failures on Murdock, I thought it was ok.

File: boards/hifive1b/include/periph_conf.h (hunk 1 above)
Comment (aabadie) [reply boards/hifive1b/include/periph_conf.h]: I dropped this change anyway.

**Comment by fjmolinas** [comment]
@aabadie you have pushed a lot to this one, is this WIP or ready for review?

**Comment by aabadie** [comment]
It's not wip anymore and should be easy to review/test if you have the hardware :)

**Comment by aabadie** [comment]
rebased (there was a conflict because #12902 was just merged)

File: cpu/fe310/periph/uart.c [hunk 2]
```diff
@@ -86,77 +83,51 @@ int uart_init(uart_t dev, uint32_t baudrate, uart_rx_cb_t rx_cb, void *arg)
...
-    if (dev == 1) {
-        for (size_t i = 0; i < len; i++) {
-            /* Wait for FIFO to empty */
-            while ((UART1_REG(UART_REG_TXFIFO) & UART_TXFIFO_FULL)
-                   == (uint32_t)UART_TXFIFO_FULL)
-                ;
+    for (size_t i = 0; i < len; i++) {
+        /* Wait for FIFO to empty */
+        while ((_REG32(uart_config[dev].addr, UART_REG_TXFIFO) & UART_TXFIFO_FULL)
+               == (uint32_t)UART_TXFIFO_FULL) {};
```
Comment (fjmolinas) [inline cpu/fe310/periph/uart.c]: Is this cast needed?

Comment (aabadie) [reply cpu/fe310/periph/uart.c]: Yes, it doesn't build without (error: comparison of integer expressions of different signedness"

**Comment by aabadie** [comment]
Ok to squash @fjmolinas ?

Regarding the code size increase, it will be greatly compensated when #12934 is merged.

File: cpu/fe310/periph/uart.c (hunk 2 above)
Comment (fjmolinas) [reply cpu/fe310/periph/uart.c]: OK!

**Comment by fjmolinas** [comment]
please squash!

**Pull request merged**